### DISCUSSION TITLE
Temporarily splitting the layout to view multiple files by holding CMD while clicking tabs

### DISCUSSION THREAD

**Discussion by nicholas-ochoa** [zed-industries/zed#23953] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

This is something Sublime Text has that I'd love to see implemented in Zed. For me, it's one of those odd features that I don't think about how often I use it until it's missing and I try to use it.

When there are multiple files open in tabs, I would like the ability to click on an inactive tab while holding down CMD and that tab will display side by side with the current tab in a vertical split layout.

If there are more than two tabs and I CMD+Click a third, it should split equally once again. There is a limit in Sublime (6-7 I think) before it stops opening new tabs.

If I hold CMD and click on an already displayed tab, it should remove that split pane from the view but keep the other already open tabs displayed.

Here is a video of the behavior from Sublime Text:

https://github.com/zed-industries/zed/assets/126299/97b901e4-2dcb-4d05-902e-69c57ad66a08

**Comment by cole-miller** [comment]
Thanks for the idea! I think this is a better fit for discussions, since we'll want to figure out whether we want this feature and if so how it should be designed.

**Comment by dkter** [comment]
Seconding this, this is the main way I use Sublime. Much less cumbersome than moving tabs around between panes.

**Comment by bienjamyn** [comment]
Wondering if this is now possible as of Sep 2026?
